libobs/media-io/video-io.c

```diff
@@ -136,6 +143,14 @@ static inline bool video_output_cur_frame(struct video_output *video)
 		struct video_input *input = video->inputs.array + i;
 		struct video_data frame = frame_info->frame;
 
+		uint32_t skip = input->frame_rate_divisor_counter++;
```

Comment:
Seems like a better implementation would be to skip `frame_rate_divisor_counter` altogether and just do

```c
if (input->frame_rate_divisor > 1 && (video->total_frames % input->frame_rate_divisor) > 0)
  continue;
```

Reply:
Potentially, though an advantage of the current implementation should be that multiple encoders started at the same time start on the same frame, as opposed to starting frames being offset by ±a few frames

Reply:
Added a comment in https://github.com/obsproject/obs-studio/pull/9001/commits/ac815904c8b387dfa0f083c5800716ca5de31fd9